pkg/cloudmap/api.go

```diff
@@ -113,12 +113,15 @@ func (sdApi *serviceDiscoveryApi) GetServiceIdMap(ctx context.Context, nsId stri
 	return serviceIdMap, nil
 }
 
-func (sdApi *serviceDiscoveryApi) DiscoverInstances(ctx context.Context, nsName string, svcName string) (insts []types.HttpInstanceSummary, err error) {
+func (sdApi *serviceDiscoveryApi) DiscoverInstances(ctx context.Context, nsName string, svcName string, clustersetId string) (insts []types.HttpInstanceSummary, err error) {
 	out, err := sdApi.awsFacade.DiscoverInstances(ctx, &sd.DiscoverInstancesInput{
 		NamespaceName: aws.String(nsName),
 		ServiceName:   aws.String(svcName),
 		HealthStatus:  types.HealthStatusFilterAll,
 		MaxResults:    aws.Int32(1000),
+		QueryParameters: map[string]string{
+			model.ClustersetIdAttr: clustersetId,
```

Comment:
needs to be `aws.String(clustersetId)`

Reply:
The function returns the wrong type expected by QueryParameters and causes type errors. The aws functions return pointer values. Also tried with StringMap(queryParams). Also tried with both combined.

Looking into the types file for DiscoverInstancesInput, the expected type is `map[string]string
` and no pointers for the map's strings either. [See here.](https://github.com/aws/aws-sdk-go-v2/blob/98519e67e1d575b00ed51b831a877a4b9bbf6a43/service/servicediscovery/api_op_DiscoverInstances.go#L70)

Errors:
`cannot use aws.String(clustersetId) (value of type *string) as string value in map literal`
`cannot use aws.StringMap((map[string]string literal)) (value of type map[string]*string) as map[string]string value in struct literal`

Reply:
FYI this was a change between sdk v1 and v2. APIGW uses v1 to call DiscoverInstances which can result in null query parameter values.